### PR TITLE
Push provisioner image for selected branches

### PR DESCRIPTION
**Pull request by andersla**
## Change content and motivation
Push provisioner image for certain branches. Creates docker image tag with branch name ('/' replaced by '-').

### CODE REVIEW

**Comment by mcapuccini** [comment]
I don't this it's a good idea, the tags page on Dockehub will become a mess. You can have this on your curated branches e.g. `phenomenal-stable`. Also could you please remove tags that are no longer needed https://hub.docker.com/r/kubenow/provisioners/tags, I think that a lot of this is not release code, but just testing and a separated repository should be used for it.

**Comment by andersla** [comment]
We could have a white-list of branches in the docker-push-section of travis which ones to push?

File: .travis.yml [hunk 1]
```diff
@@ -162,14 +167,17 @@ jobs:
...
-              docker build -t kubenow/provisioners:master .
-              docker push kubenow/provisioners:master
-            elif [ ! -z $TRAVIS_TAG ]; then
+            if [ -n "$TRAVIS_TAG" ]; then
               docker build -t kubenow/provisioners:$TRAVIS_TAG \
                 -t kubenow/provisioners:latest .
               docker push kubenow/provisioners:$TRAVIS_TAG
               docker push kubenow/provisioners:latest
+            elseif [[ "$TRAVIS_BRANCH" == "master" ||
+                      "$TRAVIS_BRANCH" == *"phenomenal"* ]]
```
Comment (mcapuccini) [inline .travis.yml]: Would be nicer to have it generic, maybe `"docker-public/"*`?

**Comment by andersla** [comment]
I found out that if you update master after opening the pull request, the "Files changed" doesn't reflect the updated master. But you can make this happen by changing base branch from master to something else, and then back again. To activate change branch dropdown menu click "Edit" on pull request: https://blog.github.com/2016-08-15-change-the-base-branch-of-a-pull-request/

**Pull request merged**